### PR TITLE
TMA-1054: alertingRules for prometheus

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1131/ : SUCCESS in 12s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/374/ : SUCCESS in 10m 27s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1137/ : SUCCESS in 12s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/380/ : SUCCESS in 6m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1139/ : SUCCESS in 11s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/382/ : SUCCESS in 15m 05s

File: k8s/charts/msf-xk8s/templates/prometheus/alertingRules.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        expr: container_pod:xk8s_pod_container_status_restarts:increase10m >= 2
+        labels:
+          severity: critical
+          team: hds # Used to reflect Icinga style alerting
+          cluster_id: {{ .Values.clusterId }}
+        annotations:
+          description: "There is more than 1 restart of {{`{{ $labels.pod }}`}} pod in the last 10 minutes"
+          summary: "{{`{{ $labels.pod }}`}} pod has too many restarts"
+      - record: "container_pod:xk8s_pod_container_status_oomkilled:increase10m"
+        expr: increase(kube_pod_container_status_terminated_reason{namespace='{{ .Release.Namespace }}', reason='OOMKilled', pod=~'^xk8s.*'}[10m])
```
Comment (sl4dy) [inline k8s/charts/msf-xk8s/templates/prometheus/alertingRules.yaml]: It makes no sense to add `pod=~'^xk8s.*'` as your pods do not have any human friendly names containing such string:
```
[root@na1-k8s-master03:/home/milan.sladky] kubectl get pods -n lcm 
NAME                                                             READY   STATUS      RESTARTS   AGE
0016a531f72f7b6b7ef7d7283095de55fa20603ad212859c00000820-66kbv   0/1     Completed   0          11d
16d9b22dbf7fbdd0c6fe7e46c15d0ed78678043cec8689ff00000820-trrwl   0/1     Completed   0          14d
1d49a420efc6ecb9bd8ad9cf7127bc97c526ccae59ff447b00000820-cwq6v   0/1     Error       0          14d
2eea2e79f2f6eb8e22ea258258a557e1e8a0c7c0de68eb7000000820-2pwdx   0/1     Completed   0          11d
414e68e0e60c0e059d34da1f8c4adb24dec3423df528260700000820-bqwfs   0/1     Completed   0          44h
b2c6e28799c8967569b0410b26f39e2ae0a1b32fba8a4bc700000820-zmlgf   0/1     Error       0          11d
b577ea144a1fe5e48c36bd1f5be401332ef1534d39c4f7f700000820-c2r56   0/1     Error       0          44h
c6c5424c3a160c764f911570435cec89fcb23475fbaf18a500000820-pwc7r   0/1     Error       0          11d
```

File: k8s/charts/msf-xk8s/templates/prometheus/alertingRules.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        expr: container_pod:xk8s_pod_container_status_oomkilled:increase10m >= 2
+        labels:
+          severity: critical
+          team: hds # Used to reflect Icinga style alerting
+          cluster_id: {{ .Values.clusterId }}
+        annotations:
+          description: "{{`{{ $labels.pod }}`}} was OOMKilled in the last 10 minutes. Investigate and/or increase memoryRequest or memoryLimit."
+          summary: "{{`{{ $labels.pod }}`}} OOMKill occured"
+      - record: "container_pod:xk8s_pod_container_status_error:increase10m"
+        expr: increase(kube_pod_container_status_terminated_reason{namespace='{{ .Release.Namespace }}', reason!~"Completed|OOMKilled", pod=~'^xk8s.*'}[10m])
```
Comment (sl4dy) [inline k8s/charts/msf-xk8s/templates/prometheus/alertingRules.yaml]: same as above

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1141/ : SUCCESS in 9s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/384/ : SUCCESS in 7m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1144/ : SUCCESS in 13s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/387/ : SUCCESS in 8m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1145/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/388/ : SUCCESS in 8m 02s

File: Jenkinsfile-chart [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (C) 2007-2019, GoodData(R) Corporation. All rights reserved.
+ */
+
+@Library('pipelines-shared-libs')
+import com.gooddata.pipeline.Pipeline
```
Comment (michalvanco) [inline Jenkinsfile-chart]: keep only following content in this file:
```
@Library('pipelines-shared-libs')
import com.gooddata.pipeline.Pipeline

Pipeline.resolve(this, '.gdc-ii-config-chart.yaml').run()
```

and add file in root `.gdc-ii-config-chart.yaml` with following content:
```
charts:
  - 'msf-xk8s'
```

This is a preliminary step before dropping the Jenkinsfiles from repositories

**Comment by michalvanco** [comment]
depends on https://github.com/gooddata/ci-infra/pull/5038

**Comment by michalvanco** [comment]
are you going to deliver this to all test/prod environments? you should define the chart in `charts-config` repository so that the "pipeline" knows where to deliver that

**Comment by michalvanco** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1146/ : SUCCESS in 9s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/389/ : SUCCESS in 7m 40s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/1/ : SUCCESS in 55s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1147/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/390/ : SUCCESS in 8m 06s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/2/ : FAILURE in 1m 01s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1148/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/391/ : SUCCESS in 8m 03s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/3/ : FAILURE in 52s

File: k8s/charts/msf-xk8s/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
```
Comment (michalvanco) [inline k8s/charts/msf-xk8s/Chart.yaml]: rename the directory `msf-xk8s` -> `lcm` (to match the chart name

**Comment by michalvanco** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/201/ : SUCCESS in 21s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-deploy-pipeline/1/ : FAILURE in 51s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/202/ : SUCCESS in 27s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-deploy-pipeline/2/ : FAILURE in 53s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/203/ : SUCCESS in 26s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-deploy-pipeline/3/ : FAILURE in 55s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1152/ : SUCCESS in 17s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/3/ : FAILURE in 19s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/5/ : FAILURE in 47s

File: Jenkinsfile-chart [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+def config = [
+        'charts': [
+                'lcm'
```
Comment (michalvanco) [inline Jenkinsfile-chart]: lcm-bricks

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1153/ : SUCCESS in 11s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/4/ : SUCCESS in 8m 34s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/6/ : SUCCESS in 1m 05s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/204/ : SUCCESS in 20s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-deploy-pipeline/4/ : FAILURE in 1m 54s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/205/ : SUCCESS in 26s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-deploy-pipeline/5/ : SUCCESS in 1m 55s

**Pull request merged**